test/vm.py

```diff
@@ -3,13 +3,42 @@
 import sys
 from io import StringIO
 
+from paramiko.client import AutoAddPolicy, SSHClient
+
 from testutil import get_free_port, wait_ssh_ready
 
 
-from paramiko.client import AutoAddPolicy, SSHClient
+class VM:
```

Comment:
(nitpick) we could use a abstract base class here but maybe I'm overthinking it

Reply:
Done.